### PR TITLE
SDN-4308: Update status merge for APBRoute and EgressFirewall.

### PR DESCRIPTION
**Pull request by npinaeva**
Update APBRoute IP validation.
Depends on https://github.com/ovn-org/ovn-kubernetes/pull/3750

### CODE REVIEW

File: bindata/network/ovn-kubernetes/common/001-crd.yaml [hunk 1]
```diff
@@ -713,14 +720,14 @@ spec:
...
                 type: array
+                x-kubernetes-list-type: set
               status:
                 description: A concise indication of whether the AdminPolicyBasedRoute
                   resource is applied with success
                 type: string
             required:
             - lastTransitionTime
             - messages
-            - status
```
Comment (jcaamano) [inline bindata/network/ovn-kubernetes/common/001-crd.yaml:723]: I talked with @npinaeva about this being a backward incompatible change. Need to confirm if we are ok with this.

File: bindata/network/ovn-kubernetes/common/002-rbac-node.yaml [hunk 2]
```diff
@@ -138,6 +138,7 @@ rules:
   resources:
   - adminpolicybasedexternalroutes
   - egressfirewalls
+  - egressfirewalls/status
```
Comment (jcaamano) [inline bindata/network/ovn-kubernetes/common/002-rbac-node.yaml:141]: Why did you need this and not  `adminpolicybasedexternalroutes/status` as well?

File: bindata/network/ovn-kubernetes/common/004-rbac-control-plane.yaml [hunk 3]
```diff
@@ -61,6 +61,10 @@ rules:
   - egressips
   - egressservices
   - egressservices/status
+  - adminpolicybasedexternalroutes
+  - adminpolicybasedexternalroutes/status
+  - egressfirewalls
+  - egressfirewalls/status
```
Comment (jcaamano) [inline bindata/network/ovn-kubernetes/common/004-rbac-control-plane.yaml:67]: I guess we are following the status quo here but we should really only need update perms on `adminpolicybasedexternalroutes/status` and `egressfirewalls/status`

File: bindata/network/ovn-kubernetes/common/002-rbac-node.yaml (hunk 2 above)
Comment (npinaeva) [reply bindata/network/ovn-kubernetes/common/002-rbac-node.yaml:141]: I actually need, but it should've already been here, because node updates status already :confused:

Comment (npinaeva) [reply bindata/network/ovn-kubernetes/common/002-rbac-node.yaml:141]: created a separate PR for that, since it is a different problem and should be backported to 4.14 https://github.com/openshift/cluster-network-operator/pull/2139

**Review by trozet — APPROVED** [review]
